libs/platform/src/lib/components/form/input-group/input-group.component.ts

```diff
@@ -0,0 +1,227 @@
+     * control status: 'success' | 'error' | 'warning' | 'default' | 'information'
+     */
+    @Input()
+    set state(status: Status) {
+        this._controlStateClass = `is-${status}`;
+        this._lastState = status;
+    }
+
+    /** Input value */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+    /** @hidden */
+    @ContentChildren(INPUT_GROUP_CHILD_TOKEN as any)
+    _children: QueryList<InputGroupInputComponent | InputGroupAddonComponent>;
+
+    /** @hidden */
+    @ContentChild(InputGroupInputComponent)
+    _input: InputGroupInputComponent;
+
+    /** @hidden */
+    @ViewChild(InputComponent, { read: ElementRef })
+    set _inputComponentElement(inputComponentElementRef: ElementRef<HTMLElement>) {
+        this._setInputElementClass(inputComponentElementRef?.nativeElement);
+    }
+
+    /** @hidden */
+    _beforeInputAddons: InputGroupAddonComponent[] = [];
+
+    /** @hidden */
+    _afterInputAddons: InputGroupAddonComponent[] = [];
+
+    /** @hidden */
+    _controlStateClass: string;
+
+    /** @hidden */
+    private _contentDensity: ContentDensity = this._inputGroupConfig.contentDensity;
+
+    /** @hidden */
+    private _lastState: Status;
+
+    /** @hidden */
+    constructor(
+        cd: ChangeDetectorRef,
+        @Optional() @Self() ngControl: NgControl,
+        @Optional() @SkipSelf() ngForm: NgForm,
+        @Optional() @SkipSelf() @Host() formField: FormField,
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>,
+        private _renderer: Renderer2,
+        protected _hostElementRef: ElementRef<HTMLElement>,
+        protected _inputGroupConfig: InputGroupConfig
+    ) {
+        super(cd, ngControl, ngForm, formField, formControl);
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        super.ngOnInit();
+        /**
+         * Assign predefined className to host element
+         * to make css selector stronger
+         */
+        this._addClassNameToHostElement(CSS_CLASS_NAME.host);
+
+        this._listenToFormErrorState();
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._listenToChildrenQueryListChanges();
+    }
+
+    /** @hidden */
+    private _listenToChildrenQueryListChanges(): void {
+        this._children.changes.pipe(startWith(this._children)).subscribe(() => {
+            this._createAddonsGroups();
+            this._setAddonsOptions();
+        });
+    }
+
+    /** @hidden */
+    private _listenToFormErrorState(): void {
+        this.stateChanges
+            .asObservable()
+            .pipe(
+                filter(() => !!this.ngControl),
+                takeUntil(this._destroyed)
+            )
+            .subscribe(() => {
+                if (this._status === this._lastState) {
+                    return;
+                }
+                this.state = this._status;
+                this._cd.markForCheck();
```

Comment:
you want to also inform parent FormField about the changes to values or state. stateChanges.

Reply:
Since this code is running in the `this.stateChanges` callback we don't need `markForCheck();` at all here, so it has been removed.

Reply:
@fkolar if I got it right we should not trigger `stateChanges` again to avoid infinite loop accidentally.
The `this.state = this._status` changes does not effect a `FormField` state, we just add a corresponding className to the html element in this case.

Reply:
Parent is trying to listen for `this.stateChanges` to trigger nessesary form field updates. 

In this case are you trying subsribe to changes from child components ?

Reply:
@fkolar  Yes, cause corresponding to the spec we need to have `@Input state: Status` defined.
In the same time `baseInput` is listening to changes and validate control assigning `error` status once it's not valid
https://github.com/SAP/fundamental-ngx/blob/8d53c3d522770a2612558095c83d0259501f214b/libs/platform/src/lib/components/form/base.input.ts#L268-L279
So I listen when it happens and assign this value to `this.state` in order to update input view correspondingly (change class name)

Reply:
@fkolar I changed a logic a bit and made `_controlStateClass` as a getter, as result I rid of `stateChanges` listening.
Please take a look.

Reply:
@fkolar 
According to the spec `input-group` should have `state` input (errors | success | info ...)
at the same time `baseInput` has `_status` property which gets set to `error` once `FormControl` is invalid
How should this behave when we use it like this?
```
<fdp-input-group state="success" [formControl]="someFormControlInInvalidState"></fdp-input-group>
```
Should we keep state value separately from `_status` and when we have `_status='error'` we show error state but when it's empty the `state` value takes precedence?

Or if `state` is not empty it should take precedence and when it's empty we return `_status` value from `baseInput`?

Reply:
@fkolar I completely removed `state` from `input-group` as it was discussed on our meeting.